crux-core/src/crux/codec.clj

```diff
@@ -229,6 +240,70 @@
                 (.putByte to (inc idx) (unchecked-byte (+ offset b)))
                 (recur (inc idx)))))))))
 
+  BigDecimal
+  (value->buffer [this ^MutableDirectBuffer to]
+    (let [this (.stripTrailingZeros this)
+          signum (.signum this)]
+
+      (-> to
+          (doto (.putByte 0 bigdec-value-type-id))
+          (doto (.putByte value-type-id-size (inc (.signum this)))))
+
+      (if (zero? signum)
+        (-> to (mem/limit-buffer (+ value-type-id-size Byte/BYTES)))
+
+        (let [bcd-len (MathCodec/putBinaryCodedDecimal (str (.abs (.unscaledValue this)))
+                                                       to
+                                                       (+ value-type-id-size Byte/BYTES Integer/BYTES))]
+          (-> to
+              (doto (.putInt (+ value-type-id-size Byte/BYTES)
+                             (MathCodec/encodeExponent this)
+                             java.nio.ByteOrder/BIG_ENDIAN))
+              (mem/limit-buffer (+ value-type-id-size Byte/BYTES Integer/BYTES bcd-len)))))))
+
+  LocalDate
+  (value->buffer [this ^MutableDirectBuffer to]
+    (-> to
+        (doto (.putByte 0 localdate-value-type-id))
+        (doto (.putInt value-type-id-size (.getYear this)))
```

Comment:
You want to use the arity taking ByteOrder/BIG_ENDIAN for anything longer than a byte (Char/Short/Int/Long etc.). Applies here and to a few other places in this PR in crux.codec.

Reply:
oh, yep, cheers :pray: